### DISCUSSION TITLE
Newtype of Pandas dataframe

### DISCUSSION THREAD

**Discussion by falan** [python/typing#1303] [Q&A] [answered]
```python
from typing import NewType

import pandas as pd

NewTypeDf = NewType("NewTypeDf", pd.DataFrame)

def test() -> NewTypeDf:
    return NewTypeDf(
       pd.DataFrame({})
    )
```
Pyright report:  **error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)**
If I change 
```python
NewTypeDf = NewType("NewTypeDf", pd.DataFrame)
```
to 
```python
NewTypeDf = NewType("NewTypeDf", str)
```
, there is no error reported.
Is this a bug of pyright or how can I typing this?

**Comment by hauntsaninja** [comment, marked as answer]
Do you have [pandas-stubs](https://pypi.org/project/pandas-stubs/) installed? If not, try installing them and see if it fixes.

**Reply by falan** [reply]
The error remains if I install the pandas-stubs.
When adding the setting of Pyright "useLibraryCodeForTypes = true", the problem is solved.
